tensorboard/components/tf_backend/test/requestManagerTests.ts

```diff
@@ -289,6 +290,83 @@ describe('backend', () => {
         // resolving the first request triggers finishTheTest
       });
     });
+
+    it('throws InvalidRequestOptionsError on bad RequestOptions', function() {
+      const rm = new RequestManager();
+        const badOptions = new RequestOptions();
+      badOptions.methodType = HttpMethodType.GET;
+      badOptions.body = "a body";
+      chai.assert.throws(
+        ()=>rm.requestWithOptions("http://www.google.com", badOptions),
+        InvalidRequestOptionsError);
+    });
+
+    describe('tests using sinon.fakeServer', function() {
+      let server;
+      
+      beforeEach(function() {
+        server = sinon.fakeServer.create();
+        server.respondImmediately = true;
+        server.respondWith("{}");
+      });
+
+      afterEach(function() {
+        server.restore();
+      });
+      
+      it('builds correct XMLHttpRequest when request(url) is called',
+         function() {
+           const rm = new RequestManager();
+           return (rm.request("my_url") as any)
+             .finally(()=>{
+               chai.assert.lengthOf(server.requests, 1);
+               chai.assert.equal(server.requests[0].url, "my_url");
+               chai.assert.equal(server.requests[0].requestBody, null);
+               chai.assert.equal(server.requests[0].method, HttpMethodType.GET);
+               chai.assert.notProperty(server.requests[0].requestHeaders,
+                                       "Content-Type");
+             }).finally(()=>{
+               server.restore();
```

Comment:
awesome that you are using sinon fake server. AFAIK, in mocha, `afterEach` gets called after every spec. Is there a reason why you had to manually call this here?

Reply:
Thanks for catching. This was anachronistic.